src/handler/global/personalstats.ts

```diff
@@ -0,0 +1,178 @@
+import { ReimbursementInfo, Skills, UserData } from "../../wrapper/models/user";
+import {
+    getUser,
+    getFormation,
+    getUnits,
+    getAllUnits,
+} from "../../wrapper/wrapper";
+import { rankMap } from "../../service/util/constants";
+import { UserSettingsRepository } from "../../repository/UserSettingsRepository";
+import { ApiError, BotError } from "../../error";
+import { makeLogger } from "../../logger";
+
+const logger = makeLogger(module);
+
+export class PersonalStats {
+    @CommandData({ type: "global" })
+    personalStats() {
+        return new SlashCommandBuilder()
+            .setName("personalstats")
+            .setDescription("Display your personal stats")
+            .toJSON();
+    }
+
+    @Command({ name: "personalstats" })
+    @Guard({ body: commandChannelGuard })
+    async stats(interaction: CommandInteraction): Promise<void> {
+        const user = await UserSettingsRepository.getUserByDiscordId(
+            interaction.user.id
+        );
+
+        if (!user || !user.valid_key || !user.api_key) {
+            throw new BotError(
+                "Please use the /start command to store  your API key in order to use this feature."
+            );
+        }
+        let userData;
+        try {
+            userData = await getUser(user.api_key);
+        } catch (e) {
+            if (e instanceof ApiError && e.code === 1) {
+                throw new BotError(
+                    "This is not a valid API key, please check your key and try again."
+                );
+            }
+            throw new BotError(
+                "Something went wrong with calling the API, please check your key and try again."
+            );
+        }
+
+        await interaction.reply({
+            embeds: [await buildStatsEmbed(userData, user.api_key)],
+        });
+    }
+}
+
+async function buildStatsEmbed(
+    user: UserData,
+    apiKey: string
+): Promise<MessageEmbed> {
+    let formation: FormationData | undefined;
+    try {
+        formation = await getFormation(apiKey);
+    } catch (e) {
+        logger.error(e);
+    }
+    const rank = rankMap.get(user.rank) ?? "";
+    const formatter = Intl.NumberFormat("en", {
+        notation: "compact",
+        maximumFractionDigits: 2,
+    });
+    const damage = formatter.format(user.personalStats.assaults.damageDealt);
+    const losses = formatter.format(
+        user.personalStats.assaults.lossesSustained
+    );
+    const destroyed = formatter.format(user.personalStats.facilities.destroyed);
+    const built = formatter.format(user.personalStats.facilities.spend);
+    return new MessageEmbed()
+        .setTitle(`${rank} ${user.name} [${user.id}]`)
+        .setDescription(
+            `${
+                formation
+                    ? getFormationStr(formation, user)
+                    : "Not currently in a Formation"
+            } | Motto: ${user.motto} | Elo: ${user.rating}`
```

Comment:
the user isn't guaranteed to have a motto

Reply:
if you dont a have motto its an empty string in the api, so it wont break or anything. or are you saying having a blank motto in the embed is stupid?

Reply:
The latter